pkg/filesystem/file.go

```diff
@@ -88,3 +88,12 @@ func FileExistsAndNotDir(path string, fs afero.Fs) bool {
 	}
 	return !info.IsDir()
 }
+
+// IsDir checks if the path provided is a directory
+func IsDir(path string, fs afero.Fs) bool {
```

Comment:
@ifbyol makes sense, see commit: bc73466c8fd3654f3f96c90ad40d0ec764890746

I had to make some changes to adapt the `getDestroyFlags` because `afero.IsDir` also returns an error, which is a condieration I took in my previous implementation not to do, but I think it's okey to have.